app/helpers/mailer_helper.rb

```diff
@@ -176,6 +180,22 @@ def style_work_tag_metadata(tags)
     "#{label}#{style_work_tag_metadata_list(tags)}".html_safe
   end
 
+  def commenter_pseud_or_name_link(comment)
+    return style_bold(comment.comment_owner_name) + style_role("Guest") unless comment.comment_owner
+
+    return style_bold("Anonymous Creator") if comment.by_anonymous_creator?
+      
+    style_link(comment.comment_owner_name, polymorphic_url(comment.comment_owner, only_path: false))
```

Comment:
I think I'm going to have to disagree with @weeklies here — I think the cleanest way to approach this is a structure like this:
```ruby
if comment.comment_owner.nil?
  [...]
elsif comment.by_anonymous_creator?
  [...]
else
  [...]
end
```
I know there's a lot of differences of opinion over when and how to use guard clauses, but I think guard clauses tend to make the most sense when they involve handling edge cases, because they allow bailing out early while skipping the main computation being performed.

In this case, I think the name of the function (`commenter_pseud_or_name_link`) indicates what the main purpose of this function is (returning the pseud link or the guest name). So only one of the three values being returned is an edge case; the other two are the main values computed by the function. I think a structure that gives them equal weight would be easier to follow.

Reply:
I agree with an `if else` depth of 1, just wanted to avoid nested `if`s previously.